engine/src/configuration/applier/hostescalation.cc

```diff
@@ -246,10 +238,11 @@ void applier::hostescalation::resolve_object(
     throw engine_error() << "Cannot find host escalations concerning host '"
                          << hostname << "'";
 
+  size_t key = hostescalation_key(obj);
   for (hostescalation_mmap::iterator it{p.first}; it != p.second; ++it) {
     /* It's a pity but for now we don't have any idea or key to verify if
      * the hostescalation is the good one. */
-    if (it->second->get_uuid() == obj.uuid()) {
+    if (it->second->internal_key() == key) {
```

Comment:
event if keys are equals, objects may be differents, a first compare of key and then a compare of objects would be better.